### PR TITLE
[docs] add Discovery Service to Database Access architecture page

### PR DESCRIPTION
**Pull request by greedy52**
Related issues:
- #23774 
- #21540

Changes:

- Updated the diagram
- Added `Discovery Service` in the list of services
- Updated `AWS RDS / Aurora` section to  `Cloud-hosted` section
- Added a `Discovery` section

New diagram shared in lucid.

### CODE REVIEW

File: docs/pages/database-access/architecture.mdx [hunk 1]
```diff
@@ -123,20 +125,34 @@ databases hosted by AWS.
...
 
-See [RDS](./guides/rds.mdx) configuration guide for details on configuring IAM
-authentication.
+The Database Service is responsible for monitoring the dynamic database
+resources including those enrolled by the Discovery Service. Once a dynamic
+database resource is added, the Database Service will start proxying it.
+
+These clouds are currently supported:
+- AWS Databases Discovery
+- Azure Databases Discovery
```
Comment (greedy52) [inline docs/pages/database-access/architecture.mdx:155]: Will add two separate guides for these, and reference the link here.

File: docs/pages/database-access/architecture.mdx [hunk 2]
```diff
@@ -36,11 +36,13 @@ In it, we have the following Teleport components:
...
 - Teleport Database Service. The "brain" that connects to the databases,
   performs database authentication and protocol parsing.
+- [Teleport Discovery Service](#discovery). Automatically discovers
+  cloud-hosted databases and enrolls them in the Teleport cluster.
 
-Database Service establishes an SSH reverse tunnel to the Proxy. As such, users
-do not need to have direct connectivity to the Database Service, or the databases
-it's connected to. As long as it can dial back to the cluster's Proxy server, it
-can be located behind a firewall.
+Database Service establishes an SSH reverse tunnel to the Proxy Service. As
```
Comment (ptgott) [inline docs/pages/database-access/architecture.mdx:42]:
```suggestion
The Database Service establishes an SSH reverse tunnel to the Proxy Service. As
```

File: docs/pages/database-access/architecture.mdx [hunk 3]
```diff
@@ -61,10 +63,10 @@ connect to a database.
 1. The user connects to the database with the `tsh db connect` command, which
    first retrieves a short-lived X.509 certificate and then launches the
    standard database client (e.g. `psql`) with this client certificate to
-   authenticate with the Teleport Proxy service.
-1. The Proxy authenticates the connection and dispatches it to the appropriate
-   Database Service based on the routing information encoded in the client
-   certificate, over the reverse tunnel.
+   authenticate with the Teleport Proxy Service.
+1. The Proxy Service authenticates the connection and dispatches it to the
+   appropriate Database Service based on the routing information encoded in the
```
Comment (ptgott) [inline docs/pages/database-access/architecture.mdx:68]:
```suggestion
   appropriate Database Service instance based on the routing information encoded in the
```

**Review by ptgott — APPROVED** [review]

**Comment by ptgott** [comment]
@greedy52 Are we backporting this?

**Comment by greedy52** [comment]
> @greedy52 Are we backporting this?

no

**Review by smallinsky — APPROVED** [review]

**Pull request merged**